src/dsl/orbit-codegen-dsl/src/test/kotlin/cloud/orbit/dsl/OrbitFileParserTest.kt

```diff
@@ -143,6 +143,134 @@ class OrbitFileParserTest {
     }
 
+    @Test
+    fun parseData_SimpleGenericField_SingleType() {
+        val text = """
+            data foo {
+                list<int32> f1 = 3;
+            }
+        """.trimIndent()
+
+        val expectedCu = CompilationUnit(
+            FAKE_PACKAGE_NAME,
+            data = listOf(
+                DataDeclaration(
+                    "foo",
+                    fields = listOf(
+                        DataField("f1", Type("list", listOf(Type("int32"))), 3)
+                    )
+                )
+            )
+        )
+
+        val actualCu = OrbitFileParser().parse(text, FAKE_PACKAGE_NAME)
+
+        Assertions.assertEquals(expectedCu, actualCu)
+    }
+
+    @Test
+    fun parseData_ComplexGenericField_SingleType() {
+        val text = """
+            data foo {
+                list<list<string>> f1 = 3;
+            }
+        """.trimIndent()
+
+        val expectedCu = CompilationUnit(
+            FAKE_PACKAGE_NAME,
+            data = listOf(
+                DataDeclaration(
+                    "foo",
+                    fields = listOf(
+                        DataField(
+                            "f1",
+                            Type(
+                                "list", listOf(
+                                    Type(
+                                        "list", listOf(
+                                            Type("string")
+                                        )
+                                    )
+                                )
+                            ),
+                            3
+                        )
+                    )
+                )
+            )
+        )
+
+        val actualCu = OrbitFileParser().parse(text, FAKE_PACKAGE_NAME)
+
+        Assertions.assertEquals(expectedCu, actualCu)
+    }
+
+    @Test
+    fun parseData_SimpleGenericField_MultipleTypes() {
+        val text = """
+            data foo {
+                map<string, int32> f1 = 3;
+            }
+        """.trimIndent()
+
+        val expectedCu = CompilationUnit(
+            FAKE_PACKAGE_NAME,
+            data = listOf(
+                DataDeclaration(
+                    "foo",
+                    fields = listOf(
+                        DataField(
+                            "f1",
+                            Type(
+                                "map", listOf(
+                                    Type("string"), Type("int32")
+                                )
+                            ),
+                            3
+                        )
+                    )
+                )
+            )
+        )
+
+        val actualCu = OrbitFileParser().parse(text, FAKE_PACKAGE_NAME)
+
+        Assertions.assertEquals(expectedCu, actualCu)
+    }
+
+    @Test
+    fun parseData_ComplexGenericField_MultipleTypes() {
+        val text = """
+            data foo {
+                map<string, list<int32>> f1 = 3;
```

Comment:
Consider adding one more test in which the generic arg is of the type of a `data` type

Reply:
I cover that in `JavaCodeGeneratorTest`. It doesn't make a difference here.